pkg/perm/perm.go

```diff
@@ -0,0 +1,61 @@
+package perm
+
+func str(a []string, f func([]string), i int) {
+	if i > len(a) {
+		f(a)
+		return
+	}
+
+	str(a, f, i+1)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '1', create a named constant for it](https://app.codacy.com/manual/afarbos/aoc/pullRequest?prid=5498712)